src/ngx_http_lua_rewriteby.c

```diff
@@ -164,24 +171,74 @@ ngx_http_lua_rewrite_handler(ngx_http_request_t *r)
 
 
 ngx_int_t
-ngx_http_lua_rewrite_handler_inline(ngx_http_request_t *r)
+ngx_http_lua_rewrite_handler_sets(ngx_http_request_t *r)
+{
+    ngx_int_t                        rc;
+    ngx_http_lua_loc_conf_t         *llcf;
+    ngx_http_lua_ctx_t              *ctx;
+    ngx_uint_t                       i;
+    ngx_http_lua_phase_handler_t    *handler;
+    ngx_array_t                     *rewrite_handlers;
+
+    dd("rewrite by lua handler sets");
+
+    rc = NGX_DECLINED;
+
+    ctx = ngx_http_get_module_ctx(r, ngx_http_lua_module);
+
+    dd("ctx = %p", ctx);
+
+    if (ctx == NULL) {
+        ctx = ngx_http_lua_create_ctx(r);
+        if (ctx == NULL) {
+            return NGX_HTTP_INTERNAL_SERVER_ERROR;
+        }
+    }
+
+    dd("entered rewrite by lua handler sets[%ld]", ctx->current_rewrite_index);
+
+    llcf = ngx_http_get_module_loc_conf(r, ngx_http_lua_module);
+
+    rewrite_handlers = llcf->rewrite_handlers;
+    handler = rewrite_handlers->elts;
+
+    for (i = ctx->current_rewrite_index; i < rewrite_handlers->nelts; ++i) {
+        ctx->current_rewrite_index = i + 1;
```

Comment:
This looks wrong to me since a single handler may yield multiple times. Do you have test cases for a single Lua handler yielding multiple times? Like a series of `ngx.sleep` calls in a single Lua handler?